charter.html

```diff
@@ -267,7 +267,7 @@ <h2>Success Criteria</h2>
     It is expected that to reach the Candidate Recommendation Snapshot stage, each normative
     specification is expected to have <a href="https://www.w3.org/Consortium/Process/#implementation-experience">at least two independent implementations</a> of every feature defined in the specification.</p>
 
-          <p>Each normative specification should contain separate sections detailing all known security and privacy implications for implementers, Web authors, and end users.</p>
+          <p>Each normative specification should contain separate sections detailing anticipated data inputs and outputs, all known security and privacy implications for implementers, Web authors, and end users.</p>
```

Comment:
A bit of a tautology here. You are saying that a normative specification should be a specification. What I am saying is that you can't have a specification without specifying these things, so why say it again?

Reply:
I'd interpret this as meaning that each specification should contain a distinct section which summarises anticipated data inputs and outputs. Not all specs actually do this; sometimes the various data used and created are discussed "across" the document. (This seems particularly likely for this group?) It would result in redundant presentations of information in each spec; in some cases this may be actively helpful, in others I suspect it would be fairly clear without.

If this proposed texts resolves any of the active concerns with the charter then I'm supportive.

Reply:
I do think that a centralized section that lists specific data inputs and outputs would be useful, especially for assessing things like privacy impact. While these may often be discussed across the document, I am wondering if it makes sense for a consistent more explicit section. Perhaps this is not the best way to phrase it.

Reply:
> Not all specs actually do this; sometimes the various data used and created are discussed "across" the document.

I think a data section would allow us to work faster. Discussing data in context, "across" the document, is useful for understanding a specific feature, but except in the cases of limited and simple features, it quickly becomes hard to see the forest for the trees. 

Given the mandate of the group, we're going to want to consider each feature's data requirements in their own right, as a whole and out of context, so that we can reason about the overall privacy exposure they represent. Having a concise, common expression of data dependencies will make it easier to consider things independent of specified use as well as across proposals.

Reply:
My concern here was this requirement as it relates to chartering.  Constraints in charter come with significant weight.  We can't decide that this is wrong later without rechartering (which is a whole thing, as we're discovering).

I have no objection to tabulating inputs and outputs in specifications if that helps us get work done, but we don't need to bake that decision into a charter.

I won't stand in the way of this if I'm the only one to object, but - on principle - charters should not proscribe anything that would be better left to the working group.

Reply:
> Constraints in charter come with significant weight.

I agree, and were we considering a domain in which data was not specifically subject to unintended uses antithetical to our goals, I would say requiring specifications to detail data inputs and outputs would burden them unnecessarily. However, given the intent and purpose of the group and the prominent focus on privacy defined in the charter:

> The purpose of these features is to support web advertising without compromising user privacy. Here "privacy" minimally refers to appropriate processing of personal information.

and given that privacy compromises are often a result of novel, unintended and unexpected uses of data, I think we would be remiss in our efforts if we didn't detail what data was employed and make specific and concerted efforts to identify and document possible abuses.